src/Workleap.Extensions.Mongo/MongoServiceCollectionExtensions.cs

```diff
@@ -37,6 +41,47 @@ public static MongoBuilder AddMongo(this IServiceCollection services, Action<Mon
         return new MongoBuilder(services);
     }
 
+    public static MongoBuilder AddCollectionConfigurations(this MongoBuilder builder, params Assembly[] assemblies)
+    {
+        var configurationTypes = assemblies.SelectMany(assembly => assembly.GetTypes()
+            .Where(t => !t.IsAbstract)
+            .Select(t => (ConcreteType: t, Interface: t.GetInterfaces().FirstOrDefault(i => i.IsMongoCollectionConfigurationInterface())))
+            .Where(t => t.Interface != null));
+
+        foreach (var configurationType in configurationTypes)
+        {
+            var (concreteType, configurationInterface) = configurationType;
```

Comment:
This method is pretty long to read, could we break it in smaller private method to help readability?

Reply:
Did the best I could